master/buildbot/data/resultspec.py

```diff
@@ -49,17 +49,26 @@ class FieldBase(object):
         'contains': lambda d, v: set(v) <= set(d),
```

Comment:
Currently the old code does not provide OR behavior. For instance:
1) https://nine.buildbot.net/api/v2/builders?tags__contains=buildbot&order=builderid
    yields multiple builders, including `builderid: 1` (first one)
2) https://nine.buildbot.net/api/v2/builders?tags__contains=try&order=builderid
    yields one builder `builderid: 1`
3) https://nine.buildbot.net/api/v2/builders?tags__contains=try&tags__contains=buildbot&order=builderid (combination of 1+2)
    yields only one builder (AND behavior), but with OR it would yield all builders shown in the queries 1 and 2

I'll create a new pull request that changes the `contains` behavior to OR and then will adapt this request accordingly.